openshift/dynamic/client.py

```diff
@@ -121,6 +121,10 @@ def get_resources_for_api_version(self, prefix, group, version, preferred):
             subresources[resource][name] = subresource
 
         for resource in resources_raw:
+            # Prevent duplicate keys
+            for key in ('prefix', 'group', 'api_version', 'client', 'preferred'):
+                resource.pop(key, None)
```

Comment:
lgtm, though I wonder if we should do something if the value doesn't match what we're providing manually. Don't know if that's going to be a real issue. Didn't think this would be either though...